src/webgpu/web_platform/copyToTexture/ImageBitmap.spec.ts

```diff
@@ -179,15 +181,79 @@ got [${failedByteActualValues.join(', ')}]`;
 
     return entry.get(color)!;
   }
+
+  // Helper functions to generate imagePixels based input configs.
+  getImagePixels(
+    format: UncompressedTextureFormat,
+    width: number,
+    height: number,
+    transparentOp: TransparentOp,
+    orientationOp: OrientationOp
+  ): Uint8ClampedArray {
+    const bytesPerPixel = kUncompressedTextureFormatInfo[format].bytesPerBlock;
+    if (bytesPerPixel === undefined) {
+      return new Uint8ClampedArray(0);
+    }
+
+    // Generate input contents by iterating 'Color' enum
+    const imagePixels = new Uint8ClampedArray(bytesPerPixel * width * height);
+    const flipYPixels = new Uint8ClampedArray(bytesPerPixel * width * height);
```

Comment:
nit: Move this constant down to where it's used (inside `if (orientationOp === 'flipY'`).

(You could alternatively modify the loop so that flipping is done inside the image generation, instead of having `flipYPixels` at all, but I don't think there's any good reason to spend time rewriting this to do that.)

Reply:
Modify the loop logic to saving the `flipYPixels`.